api-gateway/src/test/java/com/petclinic/bffapigateway/domainclientlayer/VisitsServiceClientIntegrationTest.java

```diff
@@ -133,6 +136,116 @@ void createVisitForPet_Valid() throws JsonProcessingException {
                 .verifyComplete();
 
     }
+
+    //DuplicateTime Exception Test
+    @Test
+    void createVisitForPet_DuplicateTime_ThrowsDuplicateTimeException() throws JsonProcessingException {
+        // Arrange
+        VisitRequestDTO visitRequestDTO = new VisitRequestDTO(
+                LocalDateTime.parse("2024-11-25 13:45", DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm")),
+                "Test Visit",
+                "1",
+                "2"
+        );
+
+        String errorMessage = "{\"message\":\"A visit with the same time already exists.\"}";
+        // Mock the server response for duplicate time
+        server.enqueue(new MockResponse()
+                .setResponseCode(HttpStatus.CONFLICT.value()) // 409 status
+                .setHeader(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_JSON_VALUE)
+                .setBody(errorMessage));
+
+        // Act
+        Mono<VisitResponseDTO> resultMono = visitsServiceClient.createVisitForPet(visitRequestDTO);
+
+        // Assert
+        StepVerifier.create(resultMono)
+                .expectErrorMatches(throwable -> throwable instanceof DuplicateTimeException
+                        && throwable.getMessage().contains("A visit with the same time already exists."))
```

Comment:
Could you explain why you used a `.contains` instead of an equal operation such as `.equals("message...")` or `Objects.equals(throwable.getMessage(), "A visit with the same time already exists.")`

Reply:
Really no reason why I chose .contains just a preference I had but either one works